### PR TITLE
Specify `amd64` platform to fix `ov` builds on macOS

### LINKED ISSUE #4931: Unable to run `ov build`
## Description

When running `ov build` on the latest version of the `main` branch, I run into the following error:

```
 => CACHED [1/2] FROM docker.io/library/fedora:40@sha256:b7b4b222c2a433e831c006a49a3970096  0.0s
 => ERROR [2/2] RUN mkdir /pipx   && dnf -y install dnf-plugins-core   && dnf -y config-m  28.3s
------                                                                                           
 > [2/2] RUN mkdir /pipx   && dnf -y install dnf-plugins-core   && dnf -y config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo   && dnf -y install https://dl.k6.io/rpm/repo.rpm   && dnf -y install     git     g++     just     jq     which     xdg-utils     nodejs npm     python3.12 python3.12-devel pipx     libffi-devel     docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin     unzip     postgresql postgresql-devel     k6 words   && dnf clean all   && pipx install --global     httpie     pdm pipenv      pgcli     pre-commit   && npm install -g     n     corepack   && corepack enable:
6.971 Fedora 40 - aarch64                             2.8 MB/s |  19 MB     00:06    
10.21 Fedora 40 openh264 (From Cisco) - aarch64       818  B/s | 1.4 kB     00:01    
16.11 Fedora 40 - aarch64 - Updates                   1.7 MB/s | 9.6 MB     00:05    
17.92 Dependencies resolved.
17.93 ================================================================================
17.93  Package                     Arch       Version                Repository  Size
17.93 ================================================================================
17.93 Installing:
17.93  dnf-plugins-core            noarch     4.9.0-1.fc40           updates     41 k
17.93 Installing dependencies:
17.93  dbus-libs                   aarch64    1:1.14.10-3.fc40       fedora     156 k
17.93  python3-dateutil            noarch     1:2.8.2-13.fc40        fedora     356 k
17.93  python3-dbus                aarch64    1.3.2-6.fc40           fedora     157 k
17.93  python3-distro              noarch     1.9.0-3.fc40           fedora      50 k
17.93  python3-dnf-plugins-core    noarch     4.9.0-1.fc40           updates    321 k
17.93  python3-six                 noarch     1.16.0-14.fc40         fedora      41 k
17.93  python3-systemd             aarch64    235-9.fc40             fedora     107 k
17.93  systemd-libs                aarch64    255.12-1.fc40          updates    693 k
17.93 
17.93 Transaction Summary
17.93 ================================================================================
17.93 Install  9 Packages
17.93 
17.93 Total download size: 1.9 M
17.93 Installed size: 7.2 M
17.93 Downloading Packages:
20.37 (1/9): dbus-libs-1.14.10-3.fc40.aarch64.rpm     135 kB/s | 156 kB     00:01    
20.44 (2/9): python3-dbus-1.3.2-6.fc40.aarch64.rpm    128 kB/s | 157 kB     00:01    
20.53 (3/9): python3-dateutil-2.8.2-13.fc40.noarch.rp 271 kB/s | 356 kB     00:01    
20.54 (4/9): python3-distro-1.9.0-3.fc40.noarch.rpm   288 kB/s |  50 kB     00:00    
20.71 (5/9): python3-six-1.16.0-14.fc40.noarch.rpm    151 kB/s |  41 kB     00:00    
20.76 (6/9): python3-systemd-235-9.fc40.aarch64.rpm   457 kB/s | 107 kB     00:00    
21.21 (7/9): dnf-plugins-core-4.9.0-1.fc40.noarch.rpm  62 kB/s |  41 kB     00:00    
21.82 (8/9): python3-dnf-plugins-core-4.9.0-1.fc40.no 290 kB/s | 321 kB     00:01    
23.54 (9/9): systemd-libs-255.12-1.fc40.aarch64.rpm   249 kB/s | 693 kB     00:02    
23.55 --------------------------------------------------------------------------------
23.55 Total                                           342 kB/s | 1.9 MB     00:05     
23.62 Running transaction check
23.63 Transaction check succeeded.
23.63 Running transaction test
23.65 Transaction test succeeded.
23.65 Running transaction
23.69   Preparing        :                                                        1/1 
23.71   Installing       : systemd-libs-255.12-1.fc40.aarch64                     1/9 
23.72   Installing       : dbus-libs-1:1.14.10-3.fc40.aarch64                     2/9 
23.72   Installing       : python3-dbus-1.3.2-6.fc40.aarch64                      3/9 
23.72   Installing       : python3-systemd-235-9.fc40.aarch64                     4/9 
23.73   Installing       : python3-six-1.16.0-14.fc40.noarch                      5/9 
23.73   Installing       : python3-dateutil-1:2.8.2-13.fc40.noarch                6/9 
23.73   Installing       : python3-distro-1.9.0-3.fc40.noarch                     7/9 
23.74   Installing       : python3-dnf-plugins-core-4.9.0-1.fc40.noarch           8/9 
23.74   Installing       : dnf-plugins-core-4.9.0-1.fc40.noarch                   9/9 
23.75   Running scriptlet: dnf-plugins-core-4.9.0-1.fc40.noarch                   9/9 
23.84 
23.84 Installed:
23.84   dbus-libs-1:1.14.10-3.fc40.aarch64                                            
23.84   dnf-plugins-core-4.9.0-1.fc40.noarch                                          
23.84   python3-dateutil-1:2.8.2-13.fc40.noarch                                       
23.84   python3-dbus-1.3.2-6.fc40.aarch64                                             
23.84   python3-distro-1.9.0-3.fc40.noarch                                            
23.84   python3-dnf-plugins-core-4.9.0-1.fc40.noarch                                  
23.84   python3-six-1.16.0-14.fc40.noarch                                             
23.84   python3-systemd-235-9.fc40.aarch64                                            
23.84   systemd-libs-255.12-1.fc40.aarch64                                            
23.84 
23.84 Complete!
23.94 Adding repo from: https://download.docker.com/linux/fedora/docker-ce.repo
25.38 Docker CE Stable - aarch64                       21 kB/s |  10 kB     00:00    
26.17 repo.rpm                                         40 kB/s |  10 kB     00:00    
26.32 Dependencies resolved.
26.33 ================================================================================
26.33  Package         Architecture    Version            Repository             Size
26.33 ================================================================================
26.33 Installing:
26.33  k6-rpm          noarch          0.0.2-1            @commandline           10 k
26.33 
26.33 Transaction Summary
26.33 ================================================================================
26.33 Install  1 Package
26.33 
26.33 Total size: 10 k
26.33 Installed size: 5.5 k
26.33 Downloading Packages:
26.33 Running transaction check
26.33 Transaction check succeeded.
26.33 Running transaction test
26.33 Transaction test succeeded.
26.33 Running transaction
26.34   Preparing        :                                                        1/1 
26.34   Installing       : k6-rpm-0.0.2-1.noarch                                  1/1 
26.40 
26.40 Installed:
26.40   k6-rpm-0.0.2-1.noarch                                                         
26.40 
26.40 Complete!
28.15 k6                                              190  B/s | 275  B     00:01    
28.15 Errors during downloading metadata for repository 'k6':
28.15   - Status code: 403 for https://dl.k6.io/rpm/aarch64/repodata/repomd.xml (IP: 18.172.170.14)
28.15 Error: Failed to download metadata for repo 'k6': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
28.15 Ignoring repositories: k6
28.15 Last metadata expiration check: 0:00:02 ago on Sun Sep 15 19:12:08 2024.
28.26 Package python3-3.12.2-2.fc40.aarch64 is already installed.
28.28 No match for argument: k6
28.29 Error: Unable to find a match: k6
------
Dockerfile:51
--------------------
  50 |     # - corepack: Node.js package-manager-manager
  51 | >>> RUN mkdir /pipx \
  52 | >>>   && dnf -y install dnf-plugins-core \
  53 | >>>   && dnf -y config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo \
  54 | >>>   && dnf -y install https://dl.k6.io/rpm/repo.rpm \
  55 | >>>   && dnf -y install \
  56 | >>>     git \
  57 | >>>     g++ \
  58 | >>>     just \
  59 | >>>     jq \
  60 | >>>     which \
  61 | >>>     xdg-utils \
  62 | >>>     nodejs npm \
  63 | >>>     python3.12 python3.12-devel pipx \
  64 | >>>     libffi-devel \
  65 | >>>     docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin \
  66 | >>>     unzip \
  67 | >>>     postgresql postgresql-devel \
  68 | >>>     k6 words \
  69 | >>>   && dnf clean all \
  70 | >>>   && pipx install --global \
  71 | >>>     httpie \
  72 | >>>     pdm pipenv  \
  73 | >>>     pgcli \
  74 | >>>     pre-commit \
  75 | >>>   && npm install -g \
  76 | >>>     n \
  77 | >>>     corepack \
  78 | >>>   && corepack enable
  79 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c mkdir /pipx   && dnf -y install dnf-plugins-core   && dnf -y config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo   && dnf -y install https://dl.k6.io/rpm/repo.rpm   && dnf -y install     git     g++     just     jq     which     xdg-utils     nodejs npm     python3.12 python3.12-devel pipx     libffi-devel     docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin     unzip     postgresql postgresql-devel     k6 words   && dnf clean all   && pipx install --global     httpie     pdm pipenv      pgcli     pre-commit   && npm install -g     n     corepack   && corepack enable" did not complete successfully: exit code: 1
```

## Reproduction

Run `ov build` from a fresh environment

## Environment

- Device: 15in M2 Macbook Air
- OS: MacOS Sonoma 14.6.1
- Other info: Using OrbStack

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #4931 by @d3jawu

## Description

This PR specifies `ov` to build and run a Linux amd64 image. This fixes builds breaking on macOS because the k6 RPM repo does not supply aarch64 builds which is the default architecture used unless specified with the `--platform` flag.

## Testing Instructions

1. Check out the PR.
2. Try to build `ov`: `ov build`
3. Try to initialize `ov`: `ov init`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
I re-requested the reviewers from the @WordPress/openverse-maintainers group to pick a maintainer with a Mac.

**Review by sarayourfriend — APPROVED** [review]
LGTM... but does this mean `ov` itself will run under Rosetta for macOS folks now? That's a shame if it's just to support one tool :disappointed: 

It looks like k6 doesn't publish the aarch64 builds at all in the deb or rpm packages. [This comment outlines a method for building k6 for aarch64](https://github.com/grafana/k6/issues/2136#issuecomment-1642419144), which we could use to avoid macOS folks needing to run `ov` under Rosetta.

@dhruvkb I leave it up to you. This PR is at the very least a suitable hot-fix, even if the long term solution avoids Rosetta. I'm particularly worried how this will hurt battery life for y'all on macOS, I've heard Rosetta can be rather battery hungry.

**Comment by dhruvkb** [comment]
I'll go ahead and merge this to unblock contributors, and open an issue to switch to the using the release assets and avoid Rosetta.

**Pull request merged**
